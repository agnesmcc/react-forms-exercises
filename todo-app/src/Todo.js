import React from "react";

const Todo = ({id, task, removeTodo}) => {
    return (
        <div>
            <span>{task}</span>
            <button onClick={() => removeTodo(id)}>X</button>
        </div>
    )
}

export default Todo;
