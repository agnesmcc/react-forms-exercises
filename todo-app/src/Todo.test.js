import { render } from '@testing-library/react';
import Todo from './Todo';

const TODO = <Todo id={1} task="Clean my room" />

it('renders without crashing', () => {
  const { getByText } = render(TODO);
  const task = getByText(/Clean my room/);
  expect(task).toBeInTheDocument();
})

it('matches the snapshot', () => {
  const { asFragment } = render(TODO);
  expect(asFragment()).toMatchSnapshot();
}) 
