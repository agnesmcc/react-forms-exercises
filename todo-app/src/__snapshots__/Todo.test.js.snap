// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <span>
      Clean my room
    </span>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
